### PR TITLE
Module3

### PR DESCRIPTION
**Pull request by torchik-slava**

### CODE REVIEW

**Comment by torchik-slava** [comment]
**.env** file attached in **Learn portal**

File: src/routers/userRouter.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  requestIdValidation,
+  requestBodyValidation,
+} from "../middleware";
+import userService from "../services/userService";
+
+const router = Router();
+
+router
+  .route("/")
+  .get(async (req: Request, res: Response, next: NextFunction) => {
```
Comment (aliskar) [inline src/routers/userRouter.ts:12]: It would be better to split callbacks and router for readability. 🤔

File: src/services/userService.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import userDal from "../data-access/userDal";
+import { UserAttributes, UserCreationAttributes } from "../models/userModel";
+
+const getList = async (substring?: string, limit?: string) => {
+  let users;
+  if (substring !== undefined || limit !== undefined) {
+    const isPositiveInteger = !!limit && /^\+?\d+$/.test(limit);
```
Comment (aliskar) [inline src/services/userService.ts:7]: nice check

**Review by aliskar — APPROVED** [review]
well done

**Pull request merged**